vowpalwabbit/ftrl.cc

```diff
@@ -221,22 +221,30 @@ void inner_coin_betting_update_after_prediction(ftrl_update_data& d, float x, fl
   w[W_XT] /= d.average_squared_norm_x;
 }
 
-void coin_betting_predict(ftrl& b, base_learner&, example& ec)
+bool coin_betting_predict(ftrl& b, base_learner&, example& ec)
 {
+  constexpr float x2_max = FLT_MAX;
   b.data.predict = 0;
   b.data.normalized_squared_norm_x = 0;
 
   size_t num_features_from_interactions = 0;
```

Comment:
do we want to count the num_features_from_iteractions if we are not going to use this example to update? Should lines 231 and 232 also go in the `if` statement?

Reply:
not sure.
right now, this variable is responsible for features from interactions only.
in current test there are 11 examples with 1 feature (+ 1 constant + 1 interaction).
Without change that you are proposing final example counter is 33 (although 1 example is ignored).
If we apply it, then counter drops to 32 which seems inconsistent to me - it would make sense to return 30 if we do not want to count ignored examples.
And not sure about any existing conventions what we should do in this case since there is no precedent of ignoring examples during learning as far as i know.

Reply:
ok fair enough, seems like we should leave count as is then